packages/peregrine/lib/Router/resolveUnknownRoute.js

```diff
@@ -103,12 +113,19 @@ function fetchRoute(opts) {
     url.searchParams.set('variables', JSON.stringify({ url: opts.route }));
     url.searchParams.set('operationName', 'ResolveURL');
 
+    const headers = {
+        'Content-Type': 'application/json'
+    };
+
+    // If the store is provided include it as part of the request
+    if (opts.store) {
+        headers['Store'] = opts.store;
```

Comment:
[This](https://github.com/magento/pwa-studio/blob/develop/packages/venia-concept/src/index.js#L29-L41) is how we attach auth token to all GQL requests. We could do the same for the `store` which is probably what we will end up needing to do for any queries made through ApolloClient.